sodalite-client/src/main/java/org/icgc/dcc/sodalite/client/command/Command.java

```diff
@@ -0,0 +1,108 @@
+ *                                                                                                               
+ * This program and the accompanying materials are made available under the terms of the GNU Public License v3.0.
+ * You should have received a copy of the GNU General Public License along with                                  
+ * this program. If not, see <http://www.gnu.org/licenses/>.                                                     
+ *                                                                                                               
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY                           
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES                          
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT                           
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,                                
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED                          
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;                               
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER                              
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.icgc.dcc.sodalite.client.command;
+
+import java.util.HashMap;
+import java.util.Map;
+
+//import lombok.extern.slf4j.Slf4j;
+
+//@Slf4j
+
+/**
+ * 
+ */
+public class Command {
+
+  static Map<String, Command> commands = new HashMap<String, Command>();
+  public static String[] noArgs = new String[0];
+  static {
+    commands.put("help", new HelpCommand());
+    commands.put("register", new RegisterCommand());
+    commands.put("status", new StatusCommand());
+  }
+  protected Status status;
+  String[] args;
+
+  /**
+   * @param args
+   */
+  Command(String... args) {
+    this.status = new Status();
+    this.args = args;
+  }
+
+  private void setArgs(String... args) {
+    this.args = args;
+  }
+
+  String[] getArgs() {
+    return this.args;
+  }
+
+  /***
+   * Return the command corresponding to args
+   * @param args
+   * @return If args is length 0, or the first argument is not a valid subcommand, returns an ErrorCommand with an error
+   * message.
+   * 
+   * Otherwise, returns a Command object capable of running the given subcommand.
+   */
+  public static Command parse(String[] args) {
+
+    if (args.length == 0) {
+      return new ErrorCommand("No subcommand given");
+    }
+    String cmd = args[0];
+    // log.info("Looking up command " + cmd);
+    Command c = commands.getOrDefault(cmd, new ErrorCommand("Unknown subcommand: " + cmd));
```

Comment:
val

Unless you plan or replacing all this infrastructure later with JCommander.